compiler/pragmas.nim

```diff
@@ -646,6 +646,26 @@ proc pragmaGuard(c: PContext; it: PNode; kind: TSymKind): PSym =
   else:
     result = qualifiedLookUp(c, n, {checkUndeclared})
 
+proc lookupMacro*(c: PContext, n: PNode): PSym =
+  if n.kind == nkSym:
+    result = n.sym
+    if result.kind notin {skMacro, skTemplate}: result = nil
+  else:
+    result = searchInScopes(c, considerQuotedIdent(n), {skMacro, skTemplate})
+
+proc semPragmaTempl*(c: PContext, s: PSym, n: PNode): PNode =
+  let validationNode = newNodeI(nkCall, n.info)
+  validationNode.add(newSymNode(s))
```

Comment:
's' can be the wrong symbol here, it needs to stay an unresolved ident here and then 'semExpr' performs overloading resolution properly. Or maybe it does that anyway? Check it.